superset-frontend/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -277,6 +289,17 @@ const ExploreChartPanel = ({
     ],
   );
 
+  const getDatasourceAsSaveableDataset = source => {
+    const dataset = {
+      columns: source.columns,
+      name: source?.datasource_name || 'Untitled',
+      dbId: source.database.id,
+      sql: source?.sql || '',
+      schema: source?.schema,
```

Comment:
some of these `source` objects have optional chaining and some don't?

Reply:
@eric-briscoe Wrote this in another PR and I totally stole it.